projects/igniteui-angular/src/lib/grids/filtering/excel-style/grid.excel-style-filtering.component.ts

```diff
@@ -501,7 +501,15 @@ export class IgxGridExcelStyleFilteringComponent implements OnDestroy {
     }
 
     private generateUniqueValues(columnValues: any[]) {
-        this.uniqueValues = Array.from(new Set(columnValues));
+        if (this.column.dataType === DataType.String && this.column.filteringIgnoreCase) {
+            const filteredUniqueValues = columnValues.map(s =>
+                s.toLowerCase()).reduce((map, val, i) => map.get(val) ? map : map.set(val, columnValues[i]),
+                new Map);
```

Comment:
Use optional chaining for null values.
```suggestion
            const filteredUniqueValues = columnValues.map(s => s?.toLowerCase()).
                reduce((map, val, i) => map.get(val) ? map : map.set(val, columnValues[i]),
                new Map);
```